lib/diagnostic.rb

```diff
@@ -5,44 +5,87 @@ class Diagnostic
   # Assume each entry in the list implements a length method that returns
   # a Fixnum.
   def longest(list)
-    # Your code goes here
+    ar = ['one','two','three','four','five']
+      ar.max_by(&:length) # => "three"
   end
 
+  Clarity:2
+  Comfort:2
+
+  Googled answer
+
   # Question 2
   # Write ruby code to return the sum of entries in the
   # list supplied (i.e. apply the `+` operator).
   # Assume each entry implements a `+` method.
   def sum(list)
-    # Your code goes here
+    [1, 2, 3].inject(0) {|sum, i|  sum + i }
+
   end
 
+  Clarity:3
+  Comfort:3
+
+  Googled answer
+
   # Question 3
   # Write ruby code that returns true if some of the entries
   # in the list of numbers are odd
   def some_odd(numbers)
-    # Your code goes here
+    100.times do |i|
+    next if i % 2 != 0
+    puts i
+    end
   end
 
+  Clarity:3
+  Comfort:3
+
+  Googled answer
+
   # Question 4
   # Write ruby code that returns true if every entry in the
   # list of numbers are even
   def every_even(numbers)
-    # Your code goes here
+    100.times do |i|
```

Comment:
Same here.

`numbers.any?(&:even?)`

The question also wants a true or false returned, not the number itself.